### PR TITLE
Feature/js to ts

### PR DESCRIPTION
**Pull request by KhaledHN**
What change:

1. Converting project from JS to TS
2. UI enhancements

### CODE REVIEW

File: components/Contact.tsx [hunk 1]
```diff
@@ -20,18 +32,20 @@ const Contact = () => {
           <div className="col-span-3 lg:col-span-2 w-full h-full shadow-xl shadow-gray-400 rounded-xl p-4 ">
             <div className=" lg:p-3 h-full ">
               <div>
-                <img
+                <Image
                   className=" rounded-xl hover:scale-105 ease-in duration-300 "
-                  src="contact assets/contactpic.jpg"
+                  src="/contact assets/contactpic.jpg"
```
Comment (HazemKhaled) [inline components/Contact.tsx]: Don’t use spaces in web

Also it’ll be nice to have all assets in one folder

File: components/About.tsx [hunk 2]
```diff
@@ -15,28 +16,32 @@ const About = () => {
...
+            to learn more. I started learning JavaScript and was even more
+            enthused with making websites interactive. I started freelancing for
+            e-commerce companies on the Shopify platform. Now, I am spending my
+            time building projects with React JS, Next JS, and learning new technologies.
           </p>
-          <p className="py-2 text-gray-600 underline curser-pointer">
+          <p className="py-2 text-gray-600 underline cursor-pointer">
+            <Link href="/#projects">
             Check out some of my latest projects.
+
```
Comment (HazemKhaled) [inline components/About.tsx]: Remove this empty line

File: components/About.tsx [hunk 3]
```diff
@@ -15,28 +16,32 @@ const About = () => {
...
+            </Link>
           </p>
         </div>
         <div className="w-full h-auto m-auto shadow-xl shadow-gray-400 rounded-xl flex item-center justify-center hover:scale-105 ease-in duration-300">
-          <img
+          <Image
             className="rounded-xl"
-            src="/about assets/profilePic.jpeg"
+            src="/aboutassets/ProfilePic.jpg"
             alt="/"
```
Comment (HazemKhaled) [inline components/About.tsx]: Alt should be descriptive, do you know what alt is doing?

File: components/Contact.tsx [hunk 4]
```diff
@@ -1,13 +1,25 @@
-/* eslint-disable react/no-unescaped-entities */
-/* eslint-disable @next/next/no-img-element */
 import React from "react";
 import Image from "next/image";
 import Link from "next/link";
+import { useForm, ValidationError } from "@formspree/react";
```
Comment (HazemKhaled) [inline components/Contact.tsx]: I believe this is not related to this or, please split each feature in a separate PR

File: components/Contact.tsx [hunk 5]
```diff
@@ -65,42 +79,69 @@ const Contact = () => {
...
                   <label className="uppercase text-sm py-2">Email</label>
                   <input
                     className="border-2 rounded-lg p-3 flex border-gray-300"
-                    type="email"
-                  />
+                    id="email"
+                    type="email" 
+                    name="email"            
+                    />
                 </div>
```
Comment (HazemKhaled) [inline components/Contact.tsx]: Repeated code, create input component

File: components/Main.tsx [hunk 6]
```diff
@@ -1,50 +1,51 @@
-/* eslint-disable react/no-unescaped-entities */
 import React from "react";
 import { AiOutlineMail } from "react-icons/ai";
 import { FaGithub, FaLinkedinIn } from "react-icons/fa";
 import { BsFillPersonLinesFill } from "react-icons/bs";
 import Link from "next/link";
-import Navbar from "./Navbar";
+import Navbar from "./NavBar/Navbar";
```
Comment (HazemKhaled) [inline components/Main.tsx]: It’ll better to have index and export everything there

File: components/Main.tsx [hunk 7]
```diff
@@ -1,50 +1,51 @@
...
             <div className="rounded-full shadow-lg shadow-gray-400 p-5 cursor-pointer hover:scale-110 ease-in duration-300">
               <Link href="https://github.com/KhaledHN">
-                <FaGithub />
+                <FaGithub className="text-[#5651e1]" />
               </Link>
             </div>
             <div className="rounded-full shadow-lg shadow-gray-400 p-5 cursor-pointer hover:scale-110 ease-in duration-300">
               <Link href="https://mail.google.com/mail/u/0/#sent?compose=CllgCJqXxhFbpRhxNlVPhLsSwWrlkqhLtXRKFkCtRfxVBnfFHZrnmZNmzdTStkWKskMNjXHPRhg">
-                <AiOutlineMail />
+                <AiOutlineMail className="text-[#5651e1]" />
```
Comment (HazemKhaled) [inline components/Main.tsx]: How is this related to JS to TS feature?

**Review by HazemKhaled — CHANGES_REQUESTED** [review]
I got the issue, you need to change this PR with a target to the other pending PR so I can review the changes only

File: components/About.tsx (hunk 3 above)
Comment (KhaledHN) [reply components/About.tsx]: yes alt attribute specifies an alternate text for an image, if the image cannot be displayed.

File: components/Main.tsx (hunk 6 above)
Comment (KhaledHN) [reply components/Main.tsx]: iam exporting every thing in app.js

File: components/Contact.tsx (hunk 1 above)
Comment (KhaledHN) [reply components/Contact.tsx]: its already in one folder 
all these folders is in Assets folder

File: components/Main.tsx (hunk 7 above)
Comment (KhaledHN) [reply components/Main.tsx]: this is UI enhancements

File: pages/index.js [hunk 8]
```diff
@@ -1,10 +1,10 @@
 import Head from "next/head";
 import { Inter } from "next/font/google";
-import About from "@/components/About";
+import About from "../components/About";
```
Comment (HazemKhaled) [inline pages/index.js:3]: `@` is better

**Review by HazemKhaled — CHANGES_REQUESTED** [review]
